extensions/standard-processors/tests/unit/YamlConnectionParserTest.cpp

```diff
@@ -0,0 +1,200 @@
+      SECTION("Single relationship name left empty") {
+        YAML::Node connection_node = YAML::Load(std::string {
+            "source name: \n"
+            "destination name: \n" });
+        YamlConnectionParser yaml_connection_parser(connection_node, "test_node", parent_ptr, logger);
+        // This seems incorrect, but we do not want to ruin backward compatibility
+        CHECK_NOTHROW(yaml_connection_parser.configureConnectionSourceRelationshipsFromYaml(connection));
+      }
+      SECTION("List of relationship names contains empty item") {
+        YAML::Node connection_node = YAML::Load(std::string {
+            "source relationship names:\n"
+            "- \n" });
+        YamlConnectionParser yaml_connection_parser(connection_node, "test_node", parent_ptr, logger);
+        CHECK_THROWS(yaml_connection_parser.configureConnectionSourceRelationshipsFromYaml(connection));
+      }
+      SECTION("Source and destination lookup from via id") {
+        YAML::Node connection_node = YAML::Load(std::string {
+            "source id: \n"
+            "destination id: \n" });
+        YamlConnectionParser yaml_connection_parser(connection_node, "test_node", parent_ptr, logger);
+        CHECK_THROWS(yaml_connection_parser.getSourceUUIDFromYaml());
+        CHECK_THROWS(yaml_connection_parser.getDestinationUUIDFromYaml());
+      }
+      SECTION("Source and destination lookup via name") {
+        YAML::Node connection_node = YAML::Load(std::string {
+            "source name: \n"
+            "destination name: \n" });
+        YamlConnectionParser yaml_connection_parser(connection_node, "test_node", parent_ptr, logger);
+        CHECK_THROWS(yaml_connection_parser.getSourceUUIDFromYaml());
+        CHECK_THROWS(yaml_connection_parser.getDestinationUUIDFromYaml());
+      }
+      YAML::Node connection_node = YAML::Load(std::string {
+          "max work queue size: \n"
+          "max work queue data size: \n"
+          "flowfile expiration: \n"
+          "drop empty: \n"});
+      YamlConnectionParser yaml_connection_parser(connection_node, "test_node", parent_ptr, logger);
+      CHECK_THROWS(yaml_connection_parser.getWorkQueueSizeFromYaml());
+      CHECK_THROWS(yaml_connection_parser.getWorkQueueDataSizeFromYaml());
+      CHECK_THROWS(yaml_connection_parser.getFlowFileExpirationFromYaml());
+      CHECK_THROWS(yaml_connection_parser.getDropEmptyFromYaml());
+    }
+    SECTION("With a configuration that has values of incorrect format") {
+      YAML::Node connection_node = YAML::Load(std::string {
+          "max work queue size: 2 KB\n"
+          "max work queue data size: 10 Incorrect\n"
+          "flowfile expiration: 12\n"
+          "drop empty: sup\n"});
+      YamlConnectionParser yaml_connection_parser(connection_node, "test_node", parent_ptr, logger);
+      CHECK_NOTHROW(yaml_connection_parser.getWorkQueueSizeFromYaml());
+      CHECK_NOTHROW(yaml_connection_parser.getWorkQueueDataSizeFromYaml());
```

Comment:
why are we verifying incorrect behavior?

Reply:
We need to be backward compatible with configuration files.

Reply:
I understand we would like to maintain backward compatibility, so would we only throw when the strict parsing is ready?